packages/squiggle-lang/src/js/cli/index.ts

```diff
@@ -0,0 +1,61 @@
+import fs from "fs";
+import { OutputMode, run } from "./utils";
+
+import { program } from "@commander-js/extra-typings";
+import open from "open";
+
+program
+  .command("run")
+  .arguments("[filename]")
+  .option(
+    "-e, --eval <code>",
+    "run a given squiggle code string instead of a file"
+  )
+  .option("-t --time", "output the time it took to evaluate the code")
+  .option("-q, --quiet", "don't output the results and bindings") // useful for measuring the performance or checking that the code is valid
+  .option("-b, --show-bindings", "show bindings even if the result is present") // incompatible with --quiet
+  .action((filename, options) => {
+    let src = "";
+
+    let output: OutputMode = "RESULT_OR_BINDINGS";
+    if (options.quiet && options.showBindings) {
+      program.error(
+        "--quiet and --show-bindings can't be set at the same time."
+      );
+    } else if (options.quiet) {
+      output = "NONE";
+    } else if (options.showBindings) {
+      output = "RESULT_AND_BINDINGS";
+    }
+
+    if (filename) {
+      if (options.eval) {
```

Comment:
I don't think I'm a fan of these nested if/else statements. I'd probably convert to one layer

Reply:
Yeah, thanks.

I did it this way because that's one way to satisfy typescript guards. If you check for error conditions first then Typescript can't infer that true+false and false+true are the only combinations left.

But turns out that `if (TT) ... else if (TF) ... else if (FT) ... else /* FF */ ...` works too and it's cleaner.